src/openforms/contrib/microsoft/client.py

```diff
@@ -0,0 +1,90 @@
+import json
+import os
+from io import BytesIO
+from pathlib import Path
+from typing import Union
+
+from O365 import Account
+
+from openforms.contrib.microsoft.constants import ConflictHandling
+from openforms.contrib.microsoft.exceptions import MSAuthenticationError
+from openforms.contrib.microsoft.models import MSGraphService
+
+
+class MSGraphClient:
+    """
+    wrapper to setup O365 graph client from a MSGraphService
+    """
+
+    # .default works fine for our credentials use
+    scopes = [
+        "https://graph.microsoft.com/.default",
+    ]
+
+    def __init__(self, service: MSGraphService):
+        self.service = service
+
+        self.account = Account(
+            (self.service.client_id, self.service.secret),
+            auth_flow_type="credentials",
+            tenant_id=self.service.tenant_id,
+        )
+        if not self.account.is_authenticated:
+            if not self.account.authenticate(scopes=self.scopes):
+                raise MSAuthenticationError("cannot authenticate")
+
+    @property
+    def is_authenticated(self):
+        return self.account.is_authenticated
+
+
+class MSGraphUploadHelper:
+    """
+    helper for uploading
+    - navigate to our folder in the account
+    - upload various objects in stream mode to support large files
+    """
+
+    # TODO wrap upload_()-variations in single function and auto-detect object type
+
+    def __init__(self, client: MSGraphClient):
+        self.client = client
+
+        self.storage = self.client.account.storage()
+        self.drive = self.storage.get_default_drive()
+        self.root_folder = self.drive.get_root_folder()
+
+        # lets start from root and use subfolders in the remote_path so we don't have to manage folders
+        self.target_folder = self.root_folder
+
+    def upload_disk_file(self, input_path: str, remote_path: Union[str, Path]):
+        stream_size = os.path.getsize(input_path)
+        stream = open(input_path, "rb")
+        return self.upload_stream(stream, stream_size, remote_path)
```

Comment:
use a context manager so that resources descriptors are closed when the upload is done.

```suggestion
        with open(input_path, "rb") as stream:
            return self.upload_stream(stream, stream_size, remote_path)
```